 nuplayer: Fix de-referencing NULL mSource

InstantiateDecoderAction is executed after performing reset.
By the time performReset call is over, mSource is cleared and
instantiateDecoderAction is executed on a NULL mSource - causing
crash. At this point of time, mSource is already reset and it is
not intended to be used in future, so, we can safely return.
Check for NULL to avoid crash.

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
index 85faac6..86c4814 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
@@ -1604,7 +1604,10 @@ status_t NuPlayer::instantiateDecoder(
     if (*decoder != NULL || (audio && mFlushingAudio == SHUT_DOWN)) {
         return OK;
     }
-
+    if (mSource == NULL) {
+        ALOGD("%s Ignore instantiate decoder after clearing source", __func__);
+        return INVALID_OPERATION;
+    }
     sp<AMessage> format = mSource->getFormat(audio);
 
     if (format == NULL) {
